### PR TITLE
Trio: fix host, peer, protocols, and examples

### PR DESCRIPTION
**Pull request by mhchia**
Part of the https://github.com/libp2p/py-libp2p/pull/365:
- Host
- PeerInfo
- Protocol muxer
- Examples

### CODE REVIEW

File: examples/chat/chat.py [hunk 1]
```diff
@@ -26,53 +25,48 @@
...
+            # Associate the peer with local ip address
+            await host.connect(info)
+            # Start a stream with the destination.
+            # Multiaddress of the destination peer is fetched from the peerstore using 'peerId'.
+            stream = await host.new_stream(info.peer_id, [PROTOCOL_ID])
 
-        localhost_opt = " --localhost" if localhost else ""
+            nursery.start_soon(read_data, stream)
+            nursery.start_soon(write_data, stream)
+            print("Connected to peer %s" % info.addrs[0])
```
Comment (ChihChengLiang) [inline examples/chat/chat.py]: nit: let's do f-string here

File: libp2p/__init__.py [hunk 2]
```diff
@@ -43,28 +33,27 @@ def generate_peer_id_from(key_pair: KeyPair) -> ID:
...
 def initialize_default_swarm(
-    key_pair: KeyPair,
-    id_opt: ID = None,
-    transport_opt: Sequence[str] = None,
+    key_pair: KeyPair = None,
     muxer_opt: TMuxerOptions = None,
     sec_opt: TSecurityOptions = None,
     peerstore_opt: IPeerStore = None,
-) -> Swarm:
+) -> INetworkService:
```
Comment (ChihChengLiang) [inline libp2p/__init__.py]: Why are we changing to a more abstract type to the function output here?

Comment (mhchia) [reply libp2p/__init__.py]: I intended to use interfaces over implementations. I can change it back if you think it's more reasonable.

Comment (ChihChengLiang) [reply libp2p/__init__.py]: Using interfaces sounds good 👍

Comment (pipermerriam) [reply libp2p/__init__.py]: Agreed.  This has helped us a lot in both the trinity and py-evm codebases to establish concrete interface definitions and to use them ubiquitously in our type hints.

**Comment by mhchia** [comment]
Closed by #365

**Pull request closed without merging**